### PR TITLE
Add Config to get allowed userstore types

### PR DESCRIPTION
**Pull request by emswbandara**
Fixes https://github.com/wso2/product-is/issues/8022

Following config needs to be introduced to identity.xml

```
<AllowedUserstores>
  <AllowedUserstore>org.wso2.carbon.user.core.jdbc.UniqueIDJDBCUserStoreManager</AllowedUserstore>
  <AllowedUserstore>org.wso2.carbon.user.core.jdbc.JDBCUserStoreManager</AllowedUserstore>
</AllowedUserstores>
```

Need to configure the templated config from deployment.toml as follows.

```
[[allowed_userstore]]
class = "org.wso2.carbon.user.core.jdbc.UniqueIDJDBCUserStoreManager"
[[allowed_userstore]]  
class = "org.wso2.carbon.user.core.jdbc.JDBCUserStoreManager"
```

### CODE REVIEW

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java [hunk 1]
```diff
@@ -249,7 +249,18 @@ public UserStoreDTO getUserStore(String domain) throws IdentityUserStoreMgtExcep
     @Override
     public Set<String> getAvailableUserStoreClasses() throws IdentityUserStoreMgtException {
 
-        return UserStoreManagerRegistry.getUserStoreManagerClasses();
+        return getAllowedUserstoreClasses(UserStoreManagerRegistry.getUserStoreManagerClasses());
+    }
+
+    private Set<String> getAllowedUserstoreClasses(Set<String> userstores) {
+
+        Set<String> allowedUserstores = UserStoreConfigListenersHolder.getInstance().getAllowedUserstors();
```
Comment (thanujalk) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java:257]: getAllowed**Userstors** - Typo

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java [hunk 2]
```diff
@@ -219,4 +227,36 @@ protected void unsetUserStoreConfigListenerService(UserStoreConfigListener userS
 
         UserStoreConfigListenersHolder.getInstance().unsetUserStoreConfigListenerService(userStoreConfigListener);
     }
+
+    private void readAllowedUserstoreConfiguration() {
+
+        IdentityConfigParser configParser = IdentityConfigParser.getInstance();
+        // Read allowed userstore configurations.
+        OMElement userstoresConfig = configParser.getConfigElement("AllowedUserstores");
```
Comment (thanujalk) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java:235]: Shall we introduce a constant for "AllowedUserstores"

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java [hunk 3]
```diff
@@ -219,4 +227,36 @@ protected void unsetUserStoreConfigListenerService(UserStoreConfigListener userS
...
+            while (userstoreItr.hasNext()) {
+                OMElement userstoreConfig = (OMElement) userstoreItr.next();
+                String allowedUserstore = userstoreConfig.getText();
+                if (StringUtils.isNotBlank(allowedUserstore)) {
+                    allowedUserstores.add(allowedUserstore);
+                    allowedUserstoreCount++;
+                }
+            }
+        }
+        if (!(allowedUserstoreCount > 0)) {
```
Comment (thanujalk) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java:255]: can use ==

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigListenersHolder.java [hunk 4]
```diff
@@ -69,4 +71,14 @@ public UserStoreConfigService getUserStoreConfigService() {
         return userStoreConfigService;
     }
 
+    public Set<String> getAllowedUserstors() {
```
Comment (thanujalk) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigListenersHolder.java:74]: Typo

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/pom.xml [hunk 5]
```diff
@@ -122,7 +122,8 @@
                             org.wso2.carbon.utils; version="${carbon.kernel.package.import.version.range}",
                             org.wso2.carbon.identity.base; version="${carbon.identity.package.import.version.range}",
                             org.wso2.carbon.identity.core.*; version="${carbon.identity.package.import.version.range}",
-                            org.osgi.framework; version="${osgi.framework.imp.pkg.version.range}"
+                            org.osgi.framework; version="${osgi.framework.imp.pkg.version.range}",
+                            org.apache.axiom.om
```
Comment (deshankoswatte) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/pom.xml:126]:
```suggestion
                            org.apache.axiom.om; version="${axiom.osgi.version.range}"
```
Add the version range for the package import.

**Pull request merged**